clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -603,19 +586,24 @@ public synchronized boolean hasRequest() {
                     } else
                         break;
                 }
-                if (requestEntry.getValue().isEmpty())
+                if (requests.isEmpty())
                     iterator.remove();
             }
             return expiredRequests;
         }
 
-        public synchronized List<ClientRequest> remove(Node node) {
-            List<ClientRequest> requests = unsent.remove(node);
-            return requests == null ? Collections.<ClientRequest>emptyList() : requests;
+        public synchronized ConcurrentLinkedQueue<ClientRequest> remove(Node node) {
+            ConcurrentLinkedQueue<ClientRequest> requests = unsent.remove(node);
+            return requests == null ? new ConcurrentLinkedQueue<ClientRequest>() : requests;
+        }
+
+        public synchronized Iterator<ClientRequest> requestIterator(Node node) {
+            ConcurrentLinkedQueue<ClientRequest> requests = unsent.get(node);
+            return requests == null ? null : requests.iterator();
```

Comment:
@ijuma I think @hachikuji is suggesting directly that "Maybe remove should also return null". I think it is good to return an empty iterator in `requestIterator(...)`